extjsdk/src/main/java/io/vantiq/extjsdk/InstanceConfigUtils.java

```diff
@@ -0,0 +1,187 @@
+package io.vantiq.extjsdk;
+
+import static io.vantiq.extjsdk.Utils.AUTH_TOKEN_PROPERTY_NAME;
+import static io.vantiq.extjsdk.Utils.PORT_PROPERTY_NAME;
+import static io.vantiq.extjsdk.Utils.SECRET_CREDENTIALS;
+import static io.vantiq.extjsdk.Utils.SEND_PING_PROPERTY_NAME;
+import static io.vantiq.extjsdk.Utils.SERVER_CONFIG_DIR;
+import static io.vantiq.extjsdk.Utils.SERVER_CONFIG_FILENAME;
+import static io.vantiq.extjsdk.Utils.SOURCES_PROPERTY_NAME;
+import static io.vantiq.extjsdk.Utils.TARGET_SERVER_PROPERTY_NAME;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Properties;
+
+public class InstanceConfigUtils {
+
+    // String used by methods to synch on
+    private final String SYNCH_LOCK = "synchLockString";
+    
+    public static final String CONFIG_WAS_PROGRAMMATIC = "configWasProgrammatic";
+
+    // The properties object containing the data from the server configuration file
+    private Properties serverConfigProperties;
+    
+    /**
```

Comment:
This method allows the configuration from information obtained by the connector in its own way.  The removes the dependency on a single `server.config` file